frontend/src/components/fetchSpreadsheet.ts

```diff
@@ -0,0 +1,13 @@
+import Papa from 'papaparse';
+
+export async function getData(sheet:string, rows:([])) {
```

Comment:
If `rows` is a ref, we can type it using https://vuejs.org/guide/typescript/composition-api.html#typing-ref but I'd suggest making it a function that returns a list and assigning the return value to the ref within the component, for clarity and making it easier to test in the future.

Reply:
I'm missing some nuance of the async getData function. In my latest push, the spreadsheet data initially gets assigned to the list properly, but returns empty (console log below). Do you know what needs to change here?

![image](https://github.com/Spelman-College/spelman-dashboard/assets/13266735/77622779-64e4-4f16-ad67-c6ff4820dcaa)

Reply:
the call to `Papa.parse` can happen after we return `rows`. We'll likely want to assign the output of the call to a variable and return that.

pseudo code

```
...
const out = await Papa.parse(...
...
return out
```
Of course that will change the enclosing function's signature.

Reply:
Also, the function is async so it will actually return a promise. The caller will need to wait until it resolves. See here for an example using `.then(...`

https://github.com/Spelman-College/spelman-dashboard/blob/47901eaea51914d62864320fe6b2cb2b1d327f98/frontend/src/components/data/Demo.vue#L99

Reply:
Thanks for the pointers. I restructured getData to reflect my better understanding of Promises. 

It appears that the promise is successfully fulfilled in StoriesView.vue:15, but that the PromiseResult is still undefined. Do you see anything missing in how I am resolving the Promise? 

![image](https://github.com/Spelman-College/spelman-dashboard/assets/13266735/e7f8eaaa-1a34-450f-86c6-b484902770fc)

Reply:
I apologize- I think I misled you. Async functions do return promises but since the results are set in the callback we'll need to wrap the entire call in a promise and 'resolve' it within the callback. 
```
export const getData = async (sheet:string)  => {
    return new Promise((resolve, reject) => {
        Papa.parse(sheet, {
            header: true,
            download: true,
            worker: true,
            complete (results) {
                console.log(results.data);
                resolve(results.data)
            },
        })
    })
}
```

Reply:
All set now, thanks! Sorry for the commit churn.

Views are now WAI:
![Screenshot 2023-08-23 2 08 27 PM](https://github.com/Spelman-College/spelman-dashboard/assets/13266735/4dc0cbef-0575-4391-a22a-a59f0b2fc5b4)

![Screenshot 2023-08-23 2 07 57 PM](https://github.com/Spelman-College/spelman-dashboard/assets/13266735/4c551240-a493-4ccf-b66c-536afdb40afe)